src/parsers/manifestjson.js

```diff
@@ -303,6 +303,15 @@ export default class ManifestJSONParser extends JSONParser {
       });
     }
 
+    if (
+      this.parsedJSON.browser_specific_settings &&
+      this.parsedJSON.applications &&
+      this.parsedJSON.browser_specific_settings.gecko &&
+      this.parsedJSON.applications.gecko
+    ) {
+      this.collector.addWarning(messages.DUPLICATE_GECKO_PROPERTY);
```

Comment:
Let's add the warning if both browser_specific_settings and applications are being used without specifically looking if they also include the gecko property (as we discussed and agreed on the issue).

I agree with you that we may try to find a better code for this rule, how about the following ones? 
- `MULTIPLE_BROWSER_SPECIFIC_SETTINGS`
- `DUPLICATED_BROWSER_SPECIFIC_SETTINGS`

@wagnerand do you have some other proposed rule code for this new rule that we may like more than these ones?

Reply:
@rpl not a strong preference but I am always a fan of being explicit when possible and appropriate. Maybe something along the lines of `BROWSER_SPECIFIC_SETTINGS_AND_APPLICATIONS_DEFINED`. That's a bit long of course, we could cut it down to `BROWSER_SPECIFIC_SETTINGS_APPLICATIONS` or similar.

Reply:
@wagnerand how about `REDUNDANT_APPLICATIONS_PROPERTY` or `OVERRIDDEN_APPLICATIONS_PROPERTY`?
in the end, also based on the message associated to the rule as suggested in the other comments below, the warning message is going to suggest the user to remove the redundant `"applications"` property and so the rule code would nicely match that. 
(we do not have a limit to the code anymore, we removed it recently, but if we can find a shorter one that we like then I would prefer it).

Reply:
To add to the bike-shedding, we could use `IGNORED_APPLICATIONS_PROPERTY`. I know we're not exposing that code to the user, but I like its explicitness, which could help other code contributors. Again, I don't have a super preference.

Reply:
Agree, even if extension developers will not really care about the code we do choose, I still care that we choose one that is short but also clear and explicit enough.

Personally I like your last proposed code, `IGNORED_APPLICATIONS_PROPERTY`, and I think we can proceed to use that.

@Yrkan, it looks that we do have a "winner" for the "rule code" contest ;-)
let's use `IGNORED_APPLICATIONS_PROPERTY`.

Reply:
@rpl Great `IGNORED_APPLICATIONS_PROPERTY` it is then. i'll make the modification an request another review